### PR TITLE
Mouse

### PR DESCRIPTION
**Pull request by JoelEinbinder**
#40

### CODE REVIEW

File: lib/Mouse.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    await this.up(options);
+  }
+
+  /**
+   * @param {!Object=} options
+   */
+  async down(options) {
+    if (!options)
+      options = {};
+    this._button = 'left';
```
Comment (aslushnikov) [inline lib/Mouse.js]: we should assign from options here

File: lib/FrameManager.js [hunk 2]
```diff
@@ -406,6 +409,39 @@ class Frame {
...
+
+  /**
+   * @param {string} selector
+   * @param {!Object=} options
+   * @return {!Promise}
+   */
+  async click(selector, options) {
+    await this.hover(selector);
+    await this._frameManager._mouse.press(options);
+    // This is a hack for now, to make clicking less race-prone
```
Comment (aslushnikov) [inline lib/FrameManager.js:440]: Let's file an upstream bug for the protocol about mouseUp/mouseDown events methods which return too early.

Let's refer to the bug here and explain in more details that mouse.down() and mouse.up() rely on a racy protocol methods

**Review by aslushnikov — COMMENTED** [review]
Awesome! Let's also have phantom_shim fixed

File: lib/Mouse.js (hunk 1 above)
Comment (JoelEinbinder) [reply lib/Mouse.js]: Done.

**Comment by JoelEinbinder** [comment]
Pass the phantom tests

**Pull request merged**